<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ServerPage</class>
 <widget class="QWidget" name="ServerPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>661</width>
    <height>548</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <property name="margin">
    <number>6</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Network Configuration</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_9">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QComboBox" name="netModeComboBox">
           <item>
            <property name="text">
             <string>Automatic (UPnP)</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Firewalled</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Manually Forwarded Port</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="hiddenMode_LB">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This Retroshare node is running in &amp;quot;Hidden Mode&amp;quot;. That means it can only be reached though the TOR network.&lt;/p&gt;&lt;p&gt;As such, some network options are disabled.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>[Hidden mode]</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::MinimumExpanding</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QComboBox" name="discComboBox">
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The DHT allows you to answer connection requests from your friends using BitTorrent's DHT. It greatly improves the connectivity. No information is actually stored in the DHT. It is only used as a proxy system to get in touch with other Retroshare nodes.&lt;/p&gt;&lt;p&gt;The Discovery service sends locations and PGP identities of your trusted contacts to connected peers, to help them choose new friends. The friendship is never automatic however, and both peers still need to trust each other to allow connection. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <item>
            <property name="text">
             <string>Public: DHT &amp; Discovery</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Private: Discovery Only</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Inverted: DHT Only</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Dark Net: None</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <layout class="QHBoxLayout">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <layout class="QVBoxLayout">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Local Address</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_extAddress">
               <property name="text">
                <string>External Address</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_dynDNS">
               <property name="text">
                <string>Dynamic DNS</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QGridLayout">
             <property name="spacing">
              <number>6</number>
             </property>
             <item row="0" column="0">
              <widget class="QLineEdit" name="localAddress"/>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Port:</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QSpinBox" name="localPort">
               <property name="toolTip">
                <string>Acceptable ports range from 10 to 65535. Normally Ports below 1024 are reserved by your system.</string>
               </property>
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>7812</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLineEdit" name="extAddress"/>
             </item>
             <item row="1" column="1">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Port:</string>
               </property>
              </widget>
             </item>
             <item row="1" column="2">
              <widget class="QSpinBox" name="extPort">
               <property name="toolTip">
                <string>Acceptable ports range from 10 to 65535. Normally ports below 1024 are reserved by your system.</string>
               </property>
               <property name="minimum">
                <number>10</number>
               </property>
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>7812</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLineEdit" name="dynDNS"/>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QGridLayout" name="_4">
             <property name="leftMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>6</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="iconlabel_netLimited">
               <property name="maximumSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="pixmap">
                <pixmap resource="../images.qrc">:/images/ledoff1.png</pixmap>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="textlabel_netLimited">
               <property name="text">
                <string>Local network</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QGridLayout" name="_6">
             <property name="leftMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>6</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="iconlabel_ext">
               <property name="maximumSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="pixmap">
                <pixmap resource="../images.qrc">:/images/ledoff1.png</pixmap>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="textlabel_ext">
               <property name="toolTip">
                <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The bullet turns green as soon as Retroshare manages to get your own from the websites listed below, if you enabled that action. Retroshare will also use other means to find out your own IP.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
               </property>
               <property name="text">
                <string>External ip address finder</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QGridLayout" name="_5">
             <property name="leftMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>6</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="iconlabel_upnp">
               <property name="maximumSize">
                <size>
                 <width>16</width>
                 <height>16</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
               <property name="pixmap">
                <pixmap resource="../images.qrc">:/images/ledoff1.png</pixmap>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="textlabel_upnp">
               <property name="text">
                <string>UPnP</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Known / Previous IPs:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="ipAddressList">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>150</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="showDiscStatusBar">
         <property name="text">
          <string>Show Discovery information in statusbar</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="allowIpDeterminationCB">
         <property name="toolTip">
          <string>If you uncheck this, RetroShare can only determine your IP 
when you connect to somebody. Leaving this checked helps 
connecting when you have few friends. It also helps if you're 
behind a firewall or a VPN.</string>
         </property>
         <property name="text">
          <string>Allow RetroShare to ask my ip to these websites:</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="IPServersLV">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <layout class="QHBoxLayout" name="_2">
           <item>
            <widget class="QLabel" name="label_14">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Download limit (KB/s) </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="totalDownloadRate">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This download limit covers the whole application. However, in some situations, such as when transfering many small files at once, the estimated bandwidth becomes unreliable and the total value reported by Retroshare might exceed that limit. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="suffix">
              <string>kB/s</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100000</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="_3">
           <property name="spacing">
            <number>6</number>
           </property>
           <item>
            <widget class="QLabel" name="label_15">
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
             <property name="text">
              <string>Upload limit (KB/s) </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="totalUploadRate">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The upload limit covers the entire software. Too small an upload limit might eventually block low priority services (forums, channels). A minimum recommended value is 50KB/s. &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="suffix">
              <string>kB/s</string>
             </property>
             <property name="prefix">
              <string/>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100000</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
      <zorder>showDiscStatusBar</zorder>
      <zorder>allowIpDeterminationCB</zorder>
      <zorder>IPServersLV</zorder>
      <zorder>ipAddressList</zorder>
      <zorder>label_7</zorder>
      <zorder></zorder>
      <zorder></zorder>
     </widget>
     <widget class="QWidget" name="TorTAB">
      <attribute name="title">
       <string>TOR Configuration</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QGroupBox" name="torpage_outgoing">
         <property name="title">
          <string>Outgoing TOR Connections</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>TOR Socks Proxy</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="torpage_proxyAddress"/>
              </item>
              <item>
               <widget class="QSpinBox" name="torpage_proxyPort">
                <property name="minimum">
                 <number>10</number>
                </property>
                <property name="maximum">
                 <number>65535</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QLabel" name="iconlabel_tor_outgoing">
                <property name="maximumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="pixmap">
                 <pixmap resource="../images.qrc">:/images/ledoff1.png</pixmap>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>Outgoing Okay</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>100</height>
             </size>
            </property>
            <property name="verticalScrollBarPolicy">
             <enum>Qt::ScrollBarAlwaysOff</enum>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="plainText">
             <string>TOR Socks Proxy default: 127.0.01:9050.  Set in torrc config and update here.

You can connect to Hidden Nodes, even if you 
are running a standard Node, so why not setup TOR? </string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="torpage_incoming">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Incoming TOR Connections</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <layout class="QGridLayout" name="gridLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Local Address</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="torpage_localAddress"/>
            </item>
            <item row="0" column="2">
             <widget class="QSpinBox" name="torpage_localPort">
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QLabel" name="iconlabel_tor_incoming">
                <property name="maximumSize">
                 <size>
                  <width>16</width>
                  <height>16</height>
                 </size>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="pixmap">
                 <pixmap resource="../images.qrc">:/images/ledoff1.png</pixmap>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>Incoming Okay</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Onion Address</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="torpage_onionAddress"/>
            </item>
            <item row="1" column="2">
             <widget class="QSpinBox" name="torpage_onionPort">
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>65535</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Expected torrc Port Configuration:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="torpage_configuration">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>10</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>50</height>
               </size>
              </property>
              <property name="verticalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
              <property name="plainText">
               <string>HiddenServiceDir &lt;/your/path/to/hidden/directory/service&gt;
HiddenServicePort 9191 127.0.0.1:9191</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>100</height>
             </size>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="plainText">
             <string>To Receive Connections, you must first setup a TOR Hidden Service. 
See TOR documentation for HOWTO details.

Once this is done, paste the Onion Address in the box above.
This is your external address on the TOR network.
Finally make sure that the Ports match the TOR configuration.

If you have issues connecting over TOR check the TOR logs too.</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>18</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>netModeComboBox</tabstop>
  <tabstop>discComboBox</tabstop>
  <tabstop>localAddress</tabstop>
  <tabstop>localPort</tabstop>
  <tabstop>extAddress</tabstop>
  <tabstop>extPort</tabstop>
  <tabstop>dynDNS</tabstop>
  <tabstop>showDiscStatusBar</tabstop>
 </tabstops>
 <resources>
  <include location="../images.qrc"/>
 </resources>
 <connections/>
</ui>
